src/function/scalar/string/concat.cpp

```diff
@@ -250,6 +263,11 @@ void ConcatFun::RegisterFunction(BuiltinFunctions &set) {
 
 	set.AddFunction(ScalarFunction("||", {SQLType::VARCHAR, SQLType::VARCHAR}, SQLType::VARCHAR, concat_operator));
 
+	// FIXME: there needs to be a specialized concat function for BLOB because the conventional function,
+	// i.e., '||', will cast (CastFromBlob) the two inputs from BLOB to VARCHAR, which will result in a
+	// hexadecimal string
+	set.AddFunction(ScalarFunction("concat_blob", {SQLType::BLOB, SQLType::BLOB}, SQLType::BLOB, concat_operator));
```

Comment:
One more point: this should be an overload of the function "concat", not a function with another name, no? See for example 'age.cpp' for an example of how to add multiple overloads to a function.